### PR TITLE
add intel-mila protST example

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by jiminha** [comment]
Could you provide performance/accuracy data btw GPU vs HPU for both cases that you added? 
Also, could you add test cases in tests/

**Comment by sywangyi** [comment]
image
inference performance

**Comment by sywangyi** [comment]
@jiminha please review the perf data and ci testcase

**Comment by jiminha** [comment]
@sywangyi  for zero_shot_eval, what's the criteria for passing?

File: examples/protein-folding/run_zero_shot_eval.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            attention_mask = torch.tensor(data["protein_attention_mask"], dtype=torch.long, device=device).unsqueeze(0)
+            protein_outputs = protein_model(protein_input_ids, attention_mask=attention_mask)
+            protein_feature = protein_outputs["protein_feature"]
+            protein_feature = protein_feature / protein_feature.norm(dim=-1, keepdim=True)
+            pred = logit_scale * protein_feature @ label_feature.t()
+            preds.append(pred)
+    preds = torch.cat(preds, dim=0)
+    targets = torch.tensor(targets, dtype=torch.long, device=device)
+    accuracy = (preds.argmax(dim=-1) == targets).float().mean().item()
+    logger.warning("Zero-shot accuracy: %.6f" % accuracy)
```
Comment (regisss) [inline examples/protein-folding/run_zero_shot_eval.py:100]:
```suggestion
    logger.info("Zero-shot accuracy: %.6f" % accuracy)
```

File: examples/protein-folding/run_sequence_classification.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+try:
+    from optimum.habana.utils import check_optimum_habana_min_version
+except ImportError:
+
+    def check_optimum_habana_min_version(*a, **b):
+        return ()
+
+
+# Will error if the minimal version of Optimum Habana is not installed. Remove at your own risks.
+check_optimum_habana_min_version("1.10.0")
```
Comment (regisss) [inline examples/protein-folding/run_sequence_classification.py:44]:
```suggestion
check_optimum_habana_min_version("1.11.0")
```

File: tests/test_examples.py [hunk 3]
```diff
@@ -299,12 +307,21 @@ def test(self):
...
                 self.assertEqual(return_code, 0)
+                if self.EXAMPLE_NAME == "run_zero_shot_eval":
+                    out = p.stderr.readlines()
+                    for line in out:
+                        if b"accuracy" in line:
+                            reg = re.compile("Zero-shot accuracy: (?P<accuracy>.+)")
+                            match = reg.search(line.decode("utf-8"))
+                            if match:
+                                accuracy = float(match.groupdict().get("accuracy"))
+                                self.assertTrue(accuracy > 0.43)
```
Comment (regisss) [inline tests/test_examples.py:324]: Not a big fan of parsing the output of stderr. Let's rather write the results in a JSON file in the example (you can do something similar to https://github.com/huggingface/optimum-habana/blob/9f6eba319056aec1d9806b9eb8c377b88b7ef3ce/examples/stable-diffusion/training/textual_inversion.py#L965) and parse it in the CI script.

**Review by regisss — APPROVED** [review]
